File: etamen-backend/app/Modules/Notifications/Infrastructure/Models/NotificationPreference.php
<?php

namespace App\Modules\Notifications\Infrastructure\Models;

use App\Models\User;
use App\Modules\Notifications\Domain\Enums\NotificationCategory;
use App\Modules\Notifications\Domain\Enums\NotificationChannel;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class NotificationPreference extends Model
{
    protected $fillable = [
        'user_id',
        'channel',
        'category',
        'is_enabled',
        'quiet_hours_start',
        'quiet_hours_end',
        'timezone',
        'metadata',
    ];

    protected function casts(): array
    {
        return [
            'channel' => NotificationChannel::class,
            'category' => NotificationCategory::class,
            'is_enabled' => 'boolean',
            'metadata' => 'array',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: etamen-flutter/lib/features/notifications/domain/entities/notification_preference.dart
import 'package:etamen_app/features/notifications/domain/entities/app_notification.dart';

class NotificationPreference {
  const NotificationPreference({
    this.id,
    required this.channel,
    required this.category,
    required this.isEnabled,
    this.quietHoursStart,
    this.quietHoursEnd,
    this.timezone,
    this.metadata = const {},
  });

  final int? id;
  final NotificationChannel channel;
  final NotificationCategory category;
  final bool isEnabled;
  final String? quietHoursStart;
  final String? quietHoursEnd;
  final String? timezone;
  final Map<String, dynamic> metadata;

  NotificationPreference copyWith({bool? isEnabled}) {
    return NotificationPreference(
      id: id,
      channel: channel,
      category: category,
      isEnabled: isEnabled ?? this.isEnabled,
      quietHoursStart: quietHoursStart,
      quietHoursEnd: quietHoursEnd,
      timezone: timezone,
      metadata: metadata,
    );
  }
}

enum NotificationChannel {
  inApp('in_app'),
  push('push'),
  email('email'),
  sms('sms'),
  whatsapp('whatsapp'),
  unknown('unknown');

  const NotificationChannel(this.wireValue);

  final String wireValue;

  static NotificationChannel fromWire(Object? value) {
    final normalized = value?.toString();
    return NotificationChannel.values.firstWhere(
      (item) => item.wireValue == normalized,
      orElse: () => NotificationChannel.unknown,
    );
  }
}
